### PR TITLE
Additional dataset tests

### PR DESCRIPTION
**Pull request by VincentPikand**
- Improved testing to >80% for the first time ever.
- From now on, every PR must have >80% code coverage with a lenience of 0.01%.
- Restructured folders in `test` directory.
- Restructured plotting in `src`, such that there is now a unified `visualization` folder, which also encompasses print's (check `src/counterfactuals` folder).

### CODE REVIEW

**Comment by kmariuszk** [comment]
How did you manage to remove the plotting function from code coverage? I may need to do the same thing for the R interface implementation.

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+ignore:
+  - "src/counterfactuals/visualization"
+  - "src/data_preprocessing/visualization"
+  - "src/models/visualization"
+coverage:
+  status:
+    project:
+      default:
+        target: 80%    # the required coverage value
+        threshold: 0.01%  # the leniency in hitting the target
```
Comment (laurikskl) [inline codecov.yml:10]: i dont get why threshold is needed or why it isnt 0% but you do you!

**Review by laurikskl — APPROVED** [review]
Nothing else to comment on really, tests look clean to me and the moving of plotting makes sense, quite straight forward so LGTM.

**Review by RaunoArike — APPROVED** [review]
LGTM, didn't spot any problems. Good job!

**Pull request merged**